ESP8266/ESP8266.h

```diff
@@ -28,6 +28,13 @@ class ESP8266
     ESP8266(PinName tx, PinName rx, bool debug=false);
 
     /**
+    * Check firmware version of ESP8266
+    *
+    * @return true only if ESP8266 firmware > v2
+    */
+    bool check_firmware_version(void);
```

Comment:
Hmm, thoughts on returning the firmware version and checking at a higher level? Maybe `int get_firmware_version()`?